pkg/resource/defaultclass_test.go

```diff
@@ -0,0 +1,245 @@
+								cm.Items = []corev1alpha1.ResourceClass{}
+								*o = *cm
+								return nil
+							default:
+								return errUnexpected
+							}
+						}),
+						MockStatusUpdate: test.NewMockStatusUpdateFn(nil, func(got runtime.Object) error {
+							want := &MockClaim{}
+							want.SetConditions(v1alpha1.ReconcileError(errors.New(errNoDefaultClass)))
+							if diff := cmp.Diff(want, got, test.EquateConditions()); diff != "" {
+								t.Errorf("-want, +got:\n%s", diff)
+							}
+							return nil
+						}),
+					},
+					s: MockSchemeWith(&MockClaim{}),
+				},
+				of: ClaimKind(MockGVK(&MockClaim{})),
+			},
+			want: want{result: reconcile.Result{RequeueAfter: defaultClassWait}},
+		},
+		"MultipleDefaultClasses": {
+			args: args{
+				m: &MockManager{
+					c: &test.MockClient{
+						MockGet: test.NewMockGetFn(nil, func(o runtime.Object) error {
+							switch o := o.(type) {
+							case *MockClaim:
+								cm := &MockClaim{}
+								*o = *cm
+								return nil
+							case *MockManaged:
+								return errBoom
+							default:
+								return errUnexpected
+							}
+						}),
+						MockList: test.NewMockListFn(nil, func(o runtime.Object) error {
+							switch o := o.(type) {
+							case *corev1alpha1.ResourceClassList:
+								cm := &corev1alpha1.ResourceClassList{}
+								cm.Items = []corev1alpha1.ResourceClass{
+									{},
+									{},
+								}
+								*o = *cm
+								return nil
+							default:
+								return errUnexpected
+							}
+						}),
+						MockStatusUpdate: test.NewMockStatusUpdateFn(nil, func(got runtime.Object) error {
+							want := &MockClaim{}
+							want.SetConditions(v1alpha1.ReconcileError(errors.New(errMultipleDefaultClasses)))
+							if diff := cmp.Diff(want, got, test.EquateConditions()); diff != "" {
+								t.Errorf("-want, +got:\n%s", diff)
+							}
+							return nil
+						}),
+					},
+					s: MockSchemeWith(&MockClaim{}),
+				},
+				of: ClaimKind(MockGVK(&MockClaim{})),
+			},
+			want: want{result: reconcile.Result{RequeueAfter: defaultClassWait}},
+		},
+		"Successful": {
+			args: args{
+				m: &MockManager{
+					c: &test.MockClient{
+						MockGet: test.NewMockGetFn(nil, func(o runtime.Object) error {
+							switch o := o.(type) {
+							case *MockClaim:
+								cm := &MockClaim{}
+								*o = *cm
+								return nil
+							case *MockManaged:
+								return errBoom
+							default:
+								return errUnexpected
+							}
+						}),
+						MockList: test.NewMockListFn(nil, func(o runtime.Object) error {
+							switch o := o.(type) {
+							case *corev1alpha1.ResourceClassList:
+								cm := &corev1alpha1.ResourceClassList{}
+								cm.Items = []corev1alpha1.ResourceClass{
+									{},
+								}
+								*o = *cm
+								return nil
+							default:
+								return errUnexpected
+							}
+						}),
+						MockUpdate: test.NewMockUpdateFn(nil, func(got runtime.Object) error {
+							class := corev1alpha1.ResourceClass{}
```

Comment:
Perhaps `class` should be a function level variable (i.e. declared up near `errBoom` and friends) with some object metadata (name, namespace, etc). This way we could return it in `cm.Items` above, then ensure we set the expected reference here.